Mathlib/Geometry/Manifold/Topology.lean

```diff
@@ -0,0 +1,79 @@
+/-
+Copyright (c) 2023 Michael Rothgang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Rothgang
+-/
+import Mathlib.Geometry.Manifold.SmoothManifoldWithCorners
+
+/-!
+# Topological property of topological or smooth manifolds
+In this file, we prove a few basic topological properties of manifolds.
+Let $M$ be a topological manifold (not necessarily C^n or smooth).
+* `locallyCompact_of_finiteDimensional_of_boundaryless`: If `M` is finite-dimensional, boundaryless
+  and the underlying field `𝕜` is locally compact (such as ℝ, ℂ or the p-adic numbers),
+  `M` is locally compact.
+
+**TODO:**
+* adapt the argument to include manifolds with boundary; this probably requires a
+stronger definition of boundary to show local compactness of the half-spaces
```

Comment:
I don't think you need any stronger definition: the definition of model with corner implies the fact that the range of `I` is closed (see `ModelWithCorners.closed_range`), which should be enough for all your needs. Also, the proof below is too complicated, since you can rely on [ChartedSpace.locallyCompactSpace](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Geometry/Manifold/ChartedSpace.html#ChartedSpace.locallyCompactSpace) which already contains the essence of the result.